src/Octoshift/AwsApi.cs

```diff
@@ -45,10 +45,10 @@ public virtual async Task<string> UploadToBucket(string bucketName, string fileN
         return GetPreSignedUrlForFile(bucketName, keyName);
     }
 
-    public virtual async Task<string> UploadToBucket(string bucketName, byte[] bytes, string keyName)
+    public virtual async Task<string> UploadToBucket(string bucketName, Stream content, string keyName)
     {
-        using var byteStream = new MemoryStream(bytes);
-        await _transferUtility.UploadAsync(byteStream, bucketName, keyName);
+        await _transferUtility.UploadAsync(content, bucketName, keyName);
+        content.Dispose();
```

Comment:
1. As a rule of a thumb (unless in very rare cases) who ever created a disposable resource should be responsible to dispose it, so we shouldn't dispose here at all.
2. instead of calling dispose directly, `using` should be used.

Reply:
Addressed in 82ac9f5d78b852715173f9948e0b7b7a712522bb